### PR TITLE
SW-4362 Add sub-locations to batches

### PR DESCRIPTION
**Pull request by sgrimm**
Allow seedling batches to optionally be assigned to nursery sub-locations.

Unlike accessions, a batch can be in multiple sub-locations.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1433** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1433" target="_blank">Graphite</a> 
    * **PR #1432** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1432" target="_blank">Graphite</a> 
    * **PR #1428** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1428" target="_blank">Graphite</a>  👈
      * **PR #1434** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1434" target="_blank">Graphite</a> 
        * **PR #1435** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1435" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1428?utm_source=stack-comment).

**Comment by sgrimm** [comment]
Marked this as a draft because there's still an open requirements question that might cause this to be wrong, but the implementation here covers the requirements as written.

**Comment by sgrimm** [comment]
The requirements question was answered today and it turns out this implementation (and conceptual model) is inadequate; will update the PR.

File: src/main/kotlin/com/terraformation/backend/nursery/db/BatchStore.kt [hunk 1]
```diff
@@ -731,4 +782,41 @@ class BatchStore(
...
+        .fetchSet(BATCH_SUB_LOCATIONS.SUB_LOCATION_ID.asNonNullable())
+  }
+
+  private fun updateSubLocations(
+      batchId: BatchId,
+      facilityId: FacilityId,
+      existingSubLocationIds: Set<SubLocationId>,
+      desiredSubLocationIds: Set<SubLocationId>
+  ) {
+    //    val existingSubLocationIds = fetchSubLocationIds(batchId)
```
Comment (charlie-tf) [inline src/main/kotlin/com/terraformation/backend/nursery/db/BatchStore.kt:800]: Delete comment?

**Review by charlie-tf — APPROVED** [review]
Looks good!

**Comment by sgrimm** [comment]
### Merge activity

* **Oct 27, 6:50 PM**: @sgrimm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1428).
* **Oct 27, 6:51 PM**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1428) rebased this pull request as part of a merge.
* **Oct 27, 6:51 PM**: @sgrimm merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1428).

File: src/main/kotlin/com/terraformation/backend/nursery/db/BatchStore.kt (hunk 1 above)
Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/nursery/db/BatchStore.kt:800]: Sigh, managed to forget to push this to the PR branch before merging it, but good catch and I will remove this in a separate PR.

**Pull request merged**